### ISSUE TITLE
Heresy: replace DataBuilder

### ISSUE THREAD

**Issue by bokov** [cattolyst/datafinisher#26]
[NIH](https://en.wikipedia.org/wiki/Not_invented_here) is an anti-pattern to avoid and that's why we're currently developing DF as post-processor to DataBuilder rather than a standalone app.

In the medium future, though, there may be advantages to, if not replacing, then at least incorporating DataBuilder. DataBuilder has messy and unnecessarily complicated deployment and configuration procedures and probably a bunch of extra functionality we don't need (e.g. a script specifically for installing `pip`?). We could take just the functional core of `dfbuilder.py` and make it into a lean module that does one thing, does it reliably, and isn't wrapped in a web of dynamically-named make targets.

The advantages would be:
- Could be more easily shared with non-GPC sites, meaning that DF would also become easier to disseminate
- Could be more easily automated and deployed _here_
- Tighter integration with DF
- Easier to implement new features in DataBuilder

Nor would it be that difficult, necessarily. The i2b2 interface is entirely through the JSON file, so we could just keep using the web front-end as-is. The actual work `dfbuilder.py` does amounts to pulling subsets of the `PATIENT_DIMENSION`, `OBSERVATION_FACT`, and `CONCEPT_DIMENSION` tables, and generating the `VARIABLE` table. Arguably the transformations DF does are already more complex.

**Comment by dckc** [comment]
How is dfbuilder.py "wrapped in a web of dynamically-named make targets"? The `i2b2_vm_deploy.mk` was a quick hack. At KUMC, we don't even use it.

**Comment by bokov** [comment]
I'll chalk it up to my own ignorance, Several months ago with a lot of help from xbos666 I cobbled together a long, magic of command that that invoked `i2b2_vm_deploy.mk` in a way that worked enough to give me .db files. I wrapped it in a shell script that took the name of a .json file and the name of the desired .db output file as arguments, but that's a horribly kludgy way to do it. At the time this made me think "it might be easier to just write our own than to try to figure out how this works, and while we're at it remove as many dependencies as possible". 

I've learned more about databuilder since then, xbos666 deployed it such that it now gets called by cron and processes requests for any users with permissions to use the DataBuilder plugin, and it sounds like you guys are refactoring DataBuilder for easier deployment yourselves, so our effort will be better spent helping you test it. We do want to, sorry we haven't had the cycles yet.
